sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -1023,7 +1023,12 @@ case class ShowCreateTableCommand(table: TableIdentifier) extends RunnableComman
 
       val serdeProps = metadata.storage.properties.map {
         case (key, value) =>
-          s"'${escapeSingleQuotedString(key)}' = '${escapeSingleQuotedString(value)}'"
+          val escapedValue = if (value.length == 1 && (value.head < 32 || value.head > 126)) {
```

Comment:
I think you want `Character.isISOControl` here. But this is a bit hacky as you're here hard-coding assumptions about the encoding. Why print non-printable chars, and why octal? what goes wrong?

Reply:
I need to copy an external table to another environment, but lost the create table statement.  So I want to get this create table statement by `show create table ...`, but it can't show non printable field delim.

Reply:
I see, so when the same properties are specified, they can be specified as an octal escape sequence. That makes sense to render it back that way. I'd still use `isISOControl` for better generality.